usaspending_api/bulk_download/v2/views.py

```diff
@@ -456,19 +495,19 @@ def process_filters(self, filters, award_level):
         # Get the date ranges
         try:
             date_range_filters = {}
-            if 'start_date' in filters['date_range'] and filters['date_range']['start_date']:
+            if filters['date_range']['start_date']:
                 date_range_filters['{}__gte'.format(date_attribute)] = filters['date_range']['start_date']
-            if 'end_date' in filters['date_range'] and filters['date_range']['end_date']:
+            if filters['date_range']['end_date']:
```

Comment:
What happens when a nested field isn't populated? For example, the API request has `date_range` but doesn't have the nested `start_date` and `end_date` values? Currently the values are set to `None`, that would mean the date range filters wouldn't be applied and the user could potentially get everything, right?

Reply:
Sure, I moved the argument handling beforehand in lines 232-251 (data range specifically at lines 249-251). If `date_range` isn't provided, it'll set date_range to `{'start_date': None, 'end_date': None}`. Similarly, if `date_range` is provided as an empty dictionary, it'll set it to the same. If either `start_date` or `end_date` are provided, they'll be populated accordingly. Otherwise, they're set to `None`.

Reply:
Gotcha, my original concern was they'd be able to get _everything_ without any dates set, but that isn't a possibility since we're always limiting to the first 500k records.